hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1176,24 +1176,51 @@ private OzoneFileStatus getOzoneFileStatus(OmKeyArgs args,
    */
   private OmKeyInfo createFakeDirIfShould(String volume, String bucket,
       String keyName, BucketLayout layout) throws IOException {
-    OmKeyInfo fakeDirKeyInfo = null;
     String dirKey = OzoneFSUtils.addTrailingSlashIfNeeded(keyName);
     String targetKey = OzoneFSUtils.addTrailingSlashIfNeeded(
         metadataManager.getOzoneKey(volume, bucket, keyName));
+
+    Table<String, OmKeyInfo> keyTable = metadataManager.getKeyTable(layout);
+    Iterator<Map.Entry<CacheKey<String>, CacheValue<OmKeyInfo>>> cacheIterator =
+        keyTable.cacheIterator();
+    while (cacheIterator.hasNext()) {
```

Comment:
KeyTableCache is `PartialTableCache`, which is implemented by `ConcurrentHashMap`. Will there be a problem If the iterator visits the entries out of order?
I think it's fine since the exact cache hit should be filtered in https://github.com/apache/ozone/pull/4287#issuecomment-1436492008

Reply:
The loop over the cache does not assume any order, iterates until the first match, so it should be OK.

I'm not sure about the effect on performance, it may be better or worse with this.  I'm OK with omitting this first loop completely.

Reply:
Thanks for the confirmation. Let's keep the first loop then.